mendeleev/models.py

```diff
@@ -656,6 +656,31 @@ def oxides(self) -> List[str]:
         normal_coeffs = [[str(c) if c != 1 else "" for c in t] for t in oxide_coeffs]
         return [f"{self.symbol}{cme}O{co}" for cme, co in normal_coeffs]
 
+    def __hash__(self):
+        """Custom has function to allow comparisons
+
+        This drops allt he nested, related objects since SQLAlchemy use a custom
+        unhashable `InstrumentedList`.
+        """
+        to_drop = [
+            "_ionization_energies",
+            "_oxidation_states",
+            "_sa_instance_state",
+            "_series",
+            "_series_id",
+            "ec",
+            "group",
+            "ionic_radii",
+            "isotopes",
+            "screening_constants",
+        ]
+        hashable = [(k, v) for k, v in self.__dict__.items() if k not in to_drop]
+        return hash(tuple(sorted(hashable)))
+
+    def __eq__(self, other):
+        """Overwrite the defalt comparison"""
```

Comment:
This is incorrect, two elements could compare equal by luck. Just compare the atomic number or symbol instead.

Reply:
Would you care to elaborate or give an example? I don't think I follow.

Reply:
@kalvdans it's hard for me to see how elements might compare equal by luck since the hash is calculated based on 69 core attributes of the `Element` class, which would have to exactly the same for both elements for the comparison to be true. If they are equal then I would say the elements is the same element since they share most of the attributes.

Reply:
The hash value might only be 16 bits, it is an implementation detail which we should not depend on. If some implementation have 7 bits hash values then we for sure will have a collission somewhere.

I'm not sure why you need to check all the attributes for equality. If the atomic number is the same, then all attributes are also the same.

Reply:
I couldn't find the implementation details but it doing a simple check (with python3.8 on a 64bit machine) with `bin(hash("a"))` gives me a 64 bit int which has quite a large range of values so it's very unlikely that we'll see a collision.